src/input/reader.go

```diff
@@ -0,0 +1,63 @@
+package input
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+)
+
+// UI is user-interface of input and output.
+type UI struct {
+	Interactive bool
+	Reader      io.ReadCloser
+	Writer      io.WriteCloser
+	once        sync.Once
+}
+
+var Default = &UI{}
+
+func (i *UI) init() {
+	tty, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
+	if err == nil {
+		i.Interactive = true
+		i.Reader = tty
+		i.Writer = tty
```

Comment:
This code is partially taken from my personal stash of code re-use blocks and partially simplified version of https://github.com/tcnksm/go-input and partially adapted to eventually provide http://godoc.org/github.com/miquella/ask#HiddenAsk for handling the token prompt on startup w/o echoing it.

of course i didn't make use of that in this pass so ....